src/presentation/test/integration/registration/limitedPartner/address/enter-limited-partner-principal-office-address.test.ts

```diff
@@ -0,0 +1,230 @@
+import request from "supertest";
+
+import enTranslationText from "../../../../../../../locales/en/translations.json";
+import cyTranslationText from "../../../../../../../locales/cy/translations.json";
+
+import app from "../../../app";
+
+import * as config from "config";
+import {
+  CONFIRM_LIMITED_PARTNER_PRINCIPAL_OFFICE_ADDRESS_URL,
+  ENTER_LIMITED_PARTNER_PRINCIPAL_OFFICE_ADDRESS_URL,
+  POSTCODE_LIMITED_PARTNER_PRINCIPAL_OFFICE_ADDRESS_URL,
+  TERRITORY_CHOICE_LIMITED_PARTNER_PRINCIPAL_OFFICE_ADDRESS_URL
+} from "../../../../../controller/addressLookUp/url";
+import { getUrl, setLocalesEnabled, testTranslations } from "../../../../utils";
+import AddressPageType from "../../../../../controller/addressLookUp/PageType";
+import { appDevDependencies } from "../../../../../../config/dev-dependencies";
+import LimitedPartnerBuilder, {
+  limitedPartnerLegalEntity,
+  limitedPartnerPerson
+} from "../../../../builder/LimitedPartnerBuilder";
+import { ApiErrors } from "../../../../../../domain/entities/UIErrors";
+
+describe("Enter Limited Partner Principal Office Address Page", () => {
+  const URL = getUrl(ENTER_LIMITED_PARTNER_PRINCIPAL_OFFICE_ADDRESS_URL);
+  const redirectUrl = getUrl(CONFIRM_LIMITED_PARTNER_PRINCIPAL_OFFICE_ADDRESS_URL);
+
+  beforeEach(() => {
+    setLocalesEnabled(false);
+
+    appDevDependencies.cacheRepository.feedCache(null);
+    appDevDependencies.limitedPartnerGateway.feedLimitedPartners([]);
+  });
+
+  describe("GET Enter limited partners principal office address", () => {
+    it("should load enter limited partner's principal office address page with english text", async () => {
+      setLocalesEnabled(true);
+      const limitedPartner = new LimitedPartnerBuilder()
+        .withId(appDevDependencies.limitedPartnerGateway.limitedPartnerId)
+        .isPerson()
+        .build();
+
+      appDevDependencies.limitedPartnerGateway.feedLimitedPartners([limitedPartner]);
+
+      const res = await request(app).get(URL + "?lang=en");
+
+      expect(res.status).toBe(200);
+      testTranslations(res.text, enTranslationText.address.enterAddress, [
+        "registeredOfficeAddress",
+        "principalPlaceOfBusinessAddress",
+        "jurisdictionCountry",
+        "postcodeMissing",
+        "postcodeLength",
+        "postcodeOptional",
+        "generalPartner",
+        "usualResidentialAddress"
+      ]);
+      expect(res.text).not.toContain("WELSH -");
+      expect(res.text).toContain(limitedPartnerPerson.forename?.toUpperCase());
```

Comment:
should be checking for legal entity rather than person forename

Reply:
Done.